packages/trpc/server/routers/viewer/apps.tsx

```diff
@@ -0,0 +1,237 @@
+          enabled: !input.enabled,
+        },
+      });
+
+      // Get app name from metadata
+      const localApps = getLocalAppMetadata();
+      const appMetadata = localApps.find((localApp) => localApp.slug === app.slug);
+
+      // If disabling an app then we need to alert users basesd on the app type
+      if (input.enabled) {
+        if (app.categories.some((category) => category === "calendar" || category === "video")) {
+          const appCredentials = await prisma.credential.findMany({
+            where: {
+              appId: app.slug,
+            },
+            select: {
+              user: {
+                select: {
+                  email: true,
+                  locale: true,
+                },
+              },
+            },
+          });
+
+          Promise.all(
+            appCredentials.map(async (credential) => {
+              const t = await getTranslation(credential.user?.locale || "en", "common");
+
+              if (credential.user?.email) {
+                await sendDisabledAppEmail({
+                  email: credential.user.email,
+                  appName: appMetadata?.name || app.slug,
+                  appType: app.categories,
+                  t,
+                });
+              }
+            })
+          );
+        } else {
+          const eventTypesWithApp = await prisma.eventType.findMany({
+            where: {
+              metadata: {
+                path: ["apps", app.slug as string, "enabled"],
+                equals: true,
+              },
+            },
+            select: {
+              id: true,
+              title: true,
+              users: {
+                select: {
+                  email: true,
+                  locale: true,
+                },
+              },
+              metadata: true,
+            },
+          });
+
+          // Loop through all event types and email users to alert them that payment will be paused
+          Promise.all(
+            eventTypesWithApp.map(async (eventType) => {
+              await prisma.eventType.update({
+                where: {
+                  id: eventType.id,
+                },
+                data: {
+                  metadata: {
+                    ...eventType.metadata,
+                    apps: {
+                      ...eventType.metadata.apps,
+                      [app.slug]: { ...eventType.metadata.apps[app.slug], enabled: false },
+                    },
+                  },
+                },
+              });
+
+              const t = await getTranslation(user.locale || "en", "common");
+
+              eventType.users.map(async (user) => {
+                await sendDisabledAppEmail(
+                  eventType.title,
+                  user.email,
+                  appMetadata?.name,
+                  appMetdata?.categories,
+                  eventType.id,
+                  t
+                );
+              });
+            })
+          );
+        }
```

Comment:
In the Promise.all, update the event type metadata by disabling the app and sending an email to the user